components/support/jwcrypto/src/direct.rs

```diff
@@ -0,0 +1,159 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Support for "Direct Encryption with a Shared Symmetric Key"
+use crate::{
+    aes,
+    error::{JwCryptoError, Result},
+    Algorithm, CompactJwe, EncryptionAlgorithm, JweHeader, Jwk, JwkKeyParameters,
+};
+use rc_crypto::rand;
+
+impl Jwk {
+    /// Create a new random key suitable for `Direct` symmetric encryption.
+    /// Consumers can store this for later use, probably via serde serialization.
+    pub fn new_symmetric_key(kid: Option<String>) -> Result<Self> {
```

Comment:
naming nit: `new_direct_key` would avoid introducing a new term for an existing concept.